src/staking-v3/components/Dapps.vue

```diff
@@ -49,15 +74,29 @@ export default defineComponent({
     const { getDappTier } = useDappStaking();
     const { navigateDappPage } = useDappStakingNavigation();
 
-    const dapps = computed<CombinedDappInfo[]>(() =>
-      registeredDapps.value.filter(
+    const filteredDapps = computed<(CombinedDappInfo | undefined)[]>(() => {
+      const dapps = registeredDapps.value.filter(
         (x) =>
           x.basic.mainCategory?.toLowerCase() === props.category.toLowerCase() ||
           (x.basic.mainCategory === undefined && props.category.toLowerCase() === 'others')
-      )
-    );
+      );
+
+      if (props.search === '') return dapps;
+
+      const value = props.search.toLowerCase();
+
+      const result = dapps
```

Comment:
You can simplify this logic by using `dapps.filter`

Reply:
@bobo-k2 I've updated this logic in [ccb7070](https://github.com/AstarNetwork/astar-apps/pull/1078/commits/ccb70709272e73e5af88588ba1dbcf0d0e636d66). Please review again